core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1120,6 +1120,11 @@ public List<SqlNode> fieldList() {
     }
 
     void addOrderItem(List<SqlNode> orderByList, RelFieldCollation field) {
+      // If the field being ordered on is a Literal value, we can safely skip it.
+      if (!dialect.getConformance().isSortByLiteralAllowed()
+          && field(field.getFieldIndex()) instanceof SqlLiteral) {
```

Comment:
just a question I don't know the answer to: which ordering of these checks is more "right"? would you rather check if sortByLiteral is allowed and then see if it's a literal? or does it make more sense to see if it's a literal and then check if it's allowed? 🤷

Reply:
I think especially because isSortByLiteralAllowed() is default true for all non-bq languages it's probably better to check in this order. Also not totally sure though.